src/atoms/visual-story/cover-page/cover-page.tsx

```diff
@@ -19,16 +19,53 @@ export const CoverPageBase = ({ story, config }: CoverPageProps) => {
   const headline = story.headline || "";
 
   const visualStoriesConfig = get(config, ["opts", "featureConfig", "visualStories"], {});
-  const logoAlignment =
-    typeof visualStoriesConfig.logoAlignment === "function"
-      ? visualStoriesConfig.logoAlignment(config)
-      : visualStoriesConfig.logoAlignment;
-  const logoUrl =
-    typeof visualStoriesConfig.logoUrl === "function"
-      ? visualStoriesConfig.logoUrl(config)
-      : visualStoriesConfig.logoUrl;
-  const visualStoryConfig = { logoUrl, logoAlignment };
 
+  // To support PB config
+  const isLogoAlignmentAFunction = typeof visualStoriesConfig.logoAlignment === "function"; 
+  const getLogoAlignment = isLogoAlignmentAFunction ? visualStoriesConfig.logoAlignment(config) : visualStoriesConfig.logoAlignment;
+
+  const isLogoUrlAFunction = typeof visualStoriesConfig.logoUrl === "function";
+  const getLogoUrl = isLogoUrlAFunction ? visualStoriesConfig.logoUrl(config) : visualStoriesConfig.logoUrl;
+
+  let logoAlignment = getLogoAlignment;
+  let logoUrl = getLogoUrl;
+
+  if (Array.isArray(visualStoriesConfig)) {
+    const visualStoryTheme = get(story, ["metadata", "story-attributes", "visualstorytheme"], []);
+    const theme = get(visualStoryTheme, [0]);
+    
+    // if theme 2, it will pick up value from 2nd array
+    // if theme 3, it will pick up value from 3rd array
+    // else, it will pick up from 1st array
+    const isTheme = theme === "theme-2" ? 1 : theme === "theme-3" ? 2 : 0;
+
+    // To support PB config
+    const isLogoAlignmentTheme = typeof visualStoriesConfig[isTheme].logoAlignment === "function";
+    const getLogoAlignmentTheme = isLogoAlignmentTheme ? visualStoriesConfig[isTheme].logoAlignment(config) : visualStoriesConfig[isTheme].logoAlignment;
+    
+    const isLogoUrlTheme = typeof visualStoriesConfig[isTheme].logoUrl === "function";
+    const getLogoUrlTheme = isLogoUrlTheme ? visualStoriesConfig[isTheme].logoUrl(config) : visualStoriesConfig[isTheme].logoUrl;
+   
+    switch (theme) {
+      case "theme-2": {
+        logoAlignment = visualStoriesConfig[1] && getLogoAlignmentTheme;
+        logoUrl = visualStoriesConfig[1] && getLogoUrlTheme;
+        break;
+      }
+      case "theme-3": {
+        logoAlignment = visualStoriesConfig[2] && getLogoAlignmentTheme;
+        logoUrl = visualStoriesConfig[2] && getLogoUrlTheme;
+        break;
+      }
+      default: {
+        logoAlignment = visualStoriesConfig[0] && getLogoAlignmentTheme;
+        logoUrl = visualStoriesConfig[0] && getLogoUrlTheme;
+      }
+    }
+  }
```

Comment:
Can this be written better? Seems to repeat

Reply:
Removed the switch case